wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/api/auth/oauth/OAuthAuthorizationHeaderProvider.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.auth.oauth;
+
+import javax.validation.constraints.NotNull;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Map;
+
+/**
+ * Provides the Authorization header value from the OAuth 1 providers.
+ *
+ * @author Kevin Pollet
+ */
+public interface OAuthAuthorizationHeaderProvider {
+    /**
+     * Returns the Authorization header value used to sign the request with OAuth.
+     *
+     * @param oauthProviderName
+     *         the OAuth 1 provider name.
+     * @param userId
+     *         the user id.
+     * @param requestMethod
+     *         the HTTP request method.
+     * @param requestUrl
+     *         the HTTP request url with encoded query parameters.
+     * @param requestParameters
+     *         the HTTP request parameters. HTTP request parameters must include raw values of application/x-www-form-urlencoded POST
+     *         parameters.
+     * @return the Authorization header value or {@code null} if it cannot be computed.
+     */
+    String getAuthorizationHeader(@NotNull String oauthProviderName,
+                                  @NotNull String userId,
+                                  @NotNull String requestMethod,
+                                  @NotNull String requestUrl,
+                                  @NotNull Map<String, String> requestParameters)
+            throws IOException, InvalidKeySpecException, NoSuchAlgorithmException;
```

Comment:
please add docs for exceptions. Also I'd format in following way https://github.com/eclipse/che/blob/2bd99e815e9a73b909a0f6e54958596a6a59a8cd/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java#L146-L150